torch/csrc/jit/tensorexpr/analysis.h

```diff
@@ -46,6 +47,102 @@ class NodeFinder : public IRVisitor {
   std::vector<Node*> nodes;
 };
 
+// A class that analyzes the given program relevant for Block backend.
+class BlockAnalysis : public IRVisitor {
+ public:
+  bool is_buf_store_target(const Buf* buf) const {
+    return store_targets_.count(buf) > 0;
+  }
+
+  std::unordered_set<const Buf*> reads() const {
+    return loads_;
+  }
+
+  std::unordered_set<const Buf*> stores() const {
```

Comment:
Same as above

Reply:
Thanks! This was a mistake. Returning by const ref now.